### PR TITLE
add discard htmlfield

### PR DESCRIPTION
**Pull request by FrancoisGe**

### CODE REVIEW

File: addons/html_editor/static/src/fields/html_field.js [hunk 1]
```diff
@@ -19,17 +20,43 @@ export class HtmlField extends Component {
         useBus(model.bus, "NEED_LOCAL_CHANGES", ({ detail }) =>
             detail.proms.push(this.commitChanges({ shouldInline: true }))
```
Comment (cammarosano) [inline addons/html_editor/static/src/fields/html_field.js]: `commitChanges` takes no args

Comment (FrancoisGe) [reply addons/html_editor/static/src/fields/html_field.js]: I let it to don t forget it (I need to check if we need it)

File: addons/html_editor/static/src/fields/html_field.js [hunk 2]
```diff
@@ -19,17 +20,43 @@ export class HtmlField extends Component {
...
+
+        this.isDirty = false;
+        this.state = useState({ key: 0 });
+        this.lastValue = this.props.record.data[this.props.name].toString();
+        useRecordObserver((record) => {
+            // Reset Wysiwyg when we discard
+            if (
+                !this.isDirty &&
+                !record.dirty &&
+                this.lastValue.toString() !== record.data[this.props.name].toString()
```
Comment (cammarosano) [inline addons/html_editor/static/src/fields/html_field.js]: I don't think you need to call the `toString` method here

Comment (cammarosano) [reply addons/html_editor/static/src/fields/html_field.js]: I'm talking about the one on the left

**Pull request merged**